From 471a3e41a449d7c753bc4edbba4239501bb62ba8 Mon Sep 17 00:00:00 2001
From: "D. Hugh Redelmeier" <hugh@mimosa.com>
Date: Thu, 19 Mar 2020 14:21:06 -0400
Subject: [PATCH] security: Fix for CVE-2020-1763

pluto will crash on a null pointer dereference when trying to log an error
for an IKEv1 packet containing bogus information and/or flags.

Signed-off-by: Paul Wouters <pwouters@redhat.com>
---
 programs/pluto/ikev1.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/programs/pluto/ikev1.c b/programs/pluto/ikev1.c
index 78d5c3eca1..244b00f5cc 100644
--- a/programs/pluto/ikev1.c
+++ b/programs/pluto/ikev1.c
@@ -2129,7 +2129,7 @@ void process_packet_tail(struct msg_digest *md)
 						"%smessage ignored because it contains a payload type (%s) unexpected by state %s",
 						excuse,
 						enum_show(&ikev1_payload_names, np),
-						st->st_state->name);
+						finite_states[smc->state]->name);
 					if (!md->encrypted) {
 						SEND_NOTIFICATION(INVALID_PAYLOAD_TYPE);
 					}
